indico/modules/events/registration/models/items.py

```diff
@@ -379,6 +381,18 @@ def is_visible(cls):
                    .exists())
         return cls.is_enabled & ~cls.is_deleted & ((cls.parent_id == None) | query)  # noqa
 
+    def is_locked(self, registration):
+        """Check whether the data is locked by a plugin."""
+        values = values_from_signal(signals.event.is_field_data_locked.send(self, registration=registration))
+        return any(locked for locked, _ in values)
+
+    def get_locked_reason(self, registration):
+        """Get the reason for the field being locked."""
+        values = values_from_signal(signals.event.is_field_data_locked.send(self, registration=registration))
+        for locked, reason in values:
+            if locked:
+                return reason
```

Comment:
```python
return next((reason for locked, reason in values if locked), None)
```

But maybe change the signal so it just returns a lock reason or None if it's not locked? For example what would happen right now if the signal returned `True, ''` - it'd be locked, but no useful reason would be available..